engine/src/state_chain_observer/sc_observer.rs

```diff
@@ -417,6 +416,29 @@ where
                                             let btc_rpc = btc_rpc.clone();
                                             let state_chain_client = state_chain_client.clone();
                                             scope.spawn(async move {
+                                                // We check for PendingBroadcasts for Bitcoin specifically because if the previous broadcast was not broadcast,
+                                                // it can cause ours to fail, as we could be using a change UTXO that's only created in the previous broadcast.
+                                                for awaiting_broadcast_id in state_chain_client
+                                                    .storage_value::<pallet_cf_broadcast::PendingBroadcasts<Runtime, BitcoinInstance>>(current_block.hash)
+                                                    .await?
+                                                    .into_iter()
+                                                    .filter(|b_id| *b_id < broadcast_id)
+                                                    // Sort by id to ensure we process them in order, as they may depend on each other.
+                                                    .sorted() {
+                                                        if let Some(awaiting_broadcast_data) = state_chain_client
+                                                            .storage_map_entry::<pallet_cf_broadcast::AwaitingBroadcast<Runtime, BitcoinInstance>>(current_block.hash, &awaiting_broadcast_id)
+                                                            .await? {
+                                                                match btc_rpc.send_raw_transaction(awaiting_broadcast_data.transaction_payload.encoded_transaction).await {
+                                                                    Ok(tx_hash) => info!("Bitcoin PendingBroadcast {awaiting_broadcast_id:?} success: tx_hash: {tx_hash:#x}"),
+                                                                    Err(error) => {
+                                                                        // Just log any errors, don't throw, as this one is not technically our responsibility to broadcast.
+                                                                        // and we want to get to attempting broadcasting our one.
+                                                                        error!("Error on Bitcoin PendingBroadcast {awaiting_broadcast_id:?}: {error:?}");
```

Comment:
would a `warn!` be more appropriate here? The code didn't error.

(I see this is the same in the 'normal' code path below)

Reply:
Could be a warn I guess - we don't really expect this to happen though. in the case we are submitting broadcasts, if they have already been submitted, they should return as a success - so the RPC would have errored in some way. But you're right it's not critical to the code path, so warn might be a little better. I'm indifferent, so will make the change